kernel/kernel-default/src/test/java/io/delta/kernel/parquet/TestParquetBatchReader.java

```diff
@@ -422,4 +420,35 @@ private static Tuple2<ColumnarBatch, Integer> getBatchForRowId(
 
         throw new IllegalArgumentException("row id is not found: " + rowId);
     }
+
+    @Test
+    public void requestRowIndices() {
+        String path = DefaultKernelTestUtils.getTestResourceFilePath("parquet-basic-row-indexes");
+        File dir = new File(path);
+        List<String> parquetFiles = Arrays.stream(Objects.requireNonNull(dir.listFiles()))
+                .filter(file -> file.getName().endsWith(".parquet"))
+                .map(File::getAbsolutePath)
+                .collect(Collectors.toList());
+
+        StructType readSchema = new StructType()
+                .add("id", LongType.INSTANCE)
+                .add(StructField.ROW_INDEX_COLUMN);
+
+        Configuration conf = new Configuration();
+        // Set the batch size small enough so there will be multiple batches
+        conf.setInt("delta.kernel.default.parquet.reader.batch-size", 2);
+        ParquetBatchReader reader = new ParquetBatchReader(conf);
+
+        for (String filePath : parquetFiles) {
```

Comment:
do any of these file contain multiple row groups? Probably good to add one if not. Let me know if you need the config to generate such files.

Reply:
Probably not. What's the config?

Reply:
https://github.com/delta-io/delta/blob/master/spark/src/test/scala/org/apache/spark/sql/delta/DeltaParquetFileFormatSuite.scala#L167